### ISSUE TITLE
automatic scrolling to bottom of next cell is bad when next cell is very long

### ISSUE THREAD

**Issue by brunobeltran** [jupyter/notebook#348]
Currently, in my Jupyter notebook (`pip install -U jupyter`), whenever I run a cell, the notebook automatically scrolls to the bottom of the next cell. Presumedly, this is useful if I'm running several cells in a row, but oftentimes, I'm in the situation where I have:

```
-----
small cell here
-----

-----
small output or plot here
-----

-----
HUGE cell here
-----
```

and I want to modify/run/modify/run the small cell various times. It is a HUGE hassle to have to scroll up to click on the cell I want and edit again, let along simply to see the output/plot that I was looking for. 

Maybe scrolling should first check if scrolling as it currently does would send the previous cell off-screen, and, if so, make sure that the top of the screen is instead aligned either with the top of the output of the just-executed cell or maybe some predefined (min(10% of screen height, height of 5 lines of text)) above that? 

I know that complicating the auto-scrolling effect could be undesireable, since it might violate the principle of least-confusion if we're not careful, but almost anything (even no scrolling) is better than the current behavior in the use case that I mentioned. 

This is especially annoying when making very complicated Matplotlib-based plots, as the code to generate these can often grow into 100+ lines and, especially during development, you'll want to tweak various parts of the code without separating it into a function with fixed parameters.

thoughts?

**Comment by Carreau** [comment]
Ctrl-enter run in place and does not select next cell. Does that help ?

**Comment by brunobeltran** [comment]
It helps with that particular use case, but even then, whenever the next cell is huge, scrolling to the *bottom* of it doesn't seem like the correct behavior for shift-enter. I still think that after shift-enter, for large cells, the top of the next cell (not the bottom, should be visible). 

If the idea is to minimize the amount of scrolling one has to do (so that executing puts you where you would expect) maybe control-enter should make sure the output is visible? If the current cell is large, and I'm editing some settings at the top of it, then it similarly breaks the flow of the modify/run/check output cycle if I have to scroll to see the output plot. 

The fact that with the current behavior there are cases where, no matter which execution method you use, you can't see the output without manually scrolling suggests to me that the alternative that I proposed would still improve user experience.

I realize this is work so I can jump on it myself, but I'd like feedback on whether you guys agree that this would be a positive change first.

**Comment by fonnesbeck** [comment]
I'm getting similar behavior. I've created a screen capture of it [here](http://dsh.re/18c22). For each interaction in the screen capture, I am using shift-enter. You can see that when I am in a "long" cell (i.e. one in which the output extends past the bottom of the screen), the browser view does not advance to the currently-selected cell. This makes using the notebook rather disorienting. This isn't the intended behavior, is it?

**Comment by Carreau** [comment]
From what I see most of the scrolling behavior is the native one form the browser, and there is not much we can do against it. We'll try to find ways to improve though. 

thanks.

**Comment by Burgje** [comment]
use Crtl+enter

**Comment by cipri-tom** [comment]
I have the exact opposite problem. I want to go through the cells of a notebook (`Shift + Enter`) and jupyter always goes to the one beyond view. Only after I run that one too it shifts up the new one. Ideally, I'd like it to scroll such that the top of the cell is at the top of the page

**Comment by gnestor** [comment]
I'm going to close this due to inactivity and in the interest of getting our issues under control. Feel free to open a new issue if this is not resolved.
